app/javascript/components/QuestionnaireForm/index.js

```diff
@@ -68,12 +88,247 @@ class QuestionnaireForm extends React.Component {
     }));
   }
 
+  handlePhoneChange(e) {
+    const { id } = e.target;
+    const { value } = e.target;
+    const regex = /^\s*(?:\+?(\d{1,3}))?[-. (]*(\d{3})[-. )]*(\d{3})[-. ]*(\d{4})(?: *x(\d+))?\s*$/;
+    const isValid = regex.test(value);
+    if (isValid) {
```

Comment:
can we make the field show an error only when there is something entered inside the text field and it is formatted incorrectly